packages/template-retail-react-app/app/utils/test-utils.js

```diff
@@ -182,16 +182,18 @@ export const renderWithProviders = (children, options) => {
             }
         }
     })
+    const user = userEvent.setup()
```

Comment:
As in React testing library v14.0.0, all the user actions are async, and it requires calling `setup` before performing user actions. To avoid repetitive setup, we can set it up here, right before rendering the test component, and return it along with render results, so the test can perform the user actions without having to call setup. 
```//usage
test('some test", async () => {
  const {user} = renderWithProvider(<Comp />

  await  user.click(screen.getByText*/some text/i) 
})
```